src/utils.js

```diff
@@ -45,7 +50,19 @@ function parseUrl(urlString) {
  * @return {string}
  */
 function getOrigin(loc) {
-  return loc.origin || loc.protocol + '//' + loc.host;
+  if (loc.origin) {
+    return loc.origin;
+  }
+  // Make sure that the origin is normalized. Specifically on IE, host sometimes
+  // includes the default port, which is not per standard.
```

Comment:
Why support only ports 443 and 80? Are these the only two port numbers that get wrongly parsed by IE?

Reply:
These are the only "default" ports. By URL rules, default ports must not be included in the origin. All other ports must be included.

Reply:
So to clarify: yes, only these two ports are treated incorrectly by IE.